src/hooks/useAnnouncements.tsx

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useState } from 'react'
+
+import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
+import flag from 'cozy-flags'
+import { useClient } from 'cozy-client'
+
+import {
+  Announcement,
+  AnnouncementsConfigFlag
+} from 'components/Announcements/types'
+import { filterData } from 'components/Announcements/helpers'
+import { useAnnouncementsSettings } from './useAnnouncementsSettings'
+
+interface UseAnnouncementsProps {
+  canBeDisplayed: boolean
+}
+
+const useAnnouncements = ({
+  canBeDisplayed
+}: UseAnnouncementsProps): {
+  fetchStatus: string
+  data: Announcement[]
+} => {
+  const { lang } = useI18n()
+  const client = useClient()
+
+  const [fetchStatus, setFetchStatus] = useState('pending')
+  const [rawData, setRawData] = useState<Announcement[] | null>(null)
+  const [filteredData, setFilteredData] = useState<Announcement[]>([])
+  const [hasBeenFiltered, setHasBeenFiltered] = useState(false)
+  const config = flag<AnnouncementsConfigFlag>('home.announcements')
+  const { values, save } = useAnnouncementsSettings()
+
+  useEffect(() => {
+    const fetchAnnouncements = async (): Promise<void> => {
+      setFetchStatus('loading')
+      try {
+        const resp = (await client?.stackClient.fetchJSON(
+          'GET',
+          `/remote/${config?.remoteDoctype}?lang=${lang}&channels=${config?.channels}`
+        )) as { data: Announcement[] }
+
+        setRawData(resp.data)
+        setFetchStatus('loaded')
+      } catch (error) {
+        setFetchStatus('error')
+      }
+    }
+
+    if (
+      fetchStatus === 'pending' &&
+      canBeDisplayed &&
+      config?.remoteDoctype &&
+      config?.channels
+    ) {
+      void fetchAnnouncements()
+    }
+  }, [client?.stackClient, config, fetchStatus, lang, values, canBeDisplayed])
+
+  useEffect(() => {
+    if (rawData !== null && values && !hasBeenFiltered) {
+      const uuidSeen = values.seen ?? []
+      const uuidFromApi = rawData.map(({ attributes }) => attributes.uuid)
+
+      // we only keep the announcements seen that are still returned by the API
+      // to limit the size of the seen array
+      const uuidInCommon = uuidSeen.filter(
+        (uuid: string) => uuidFromApi.indexOf(uuid) !== -1
```

Comment:
nit: this one is more adapted to ne need
```suggestion
        (uuid: string) => uuidFromApi.includes(uuid)
```